cloudinit/sources/DataSourceGCE.py

```diff
@@ -88,13 +88,39 @@ def __init__(self, sys_cfg, distro, paths):
 
     def _get_data(self):
         url_params = self.get_url_params()
-        network_context = noop()
         if self.perform_dhcp_setup:
-            network_context = EphemeralDHCPv4(
-                self.distro,
-                self.fallback_interface,
+            candidate_nics = net.find_candidate_nics(
+                default_primary_interface=DEFAULT_PRIMARY_INTERFACE
```

Comment:
Let's debug log the candidate NICs we are going to try iterating through. Just because we may glean something interesting on bigger multi-NIC instances that sheds light on ordering/priority problems past the default ens4 NIC.

Also do we have a cloud-init log of the get_data operation working on the affected system. I'd also be curious if we don't can avoid passing DEFAULT_PRIMARY_INTERFACE on this instance type and `cloud-init clean --logs ---reboot` to see sort of delays we can expect on timeouts for invalid NICs prior to finding the valid ens4.